File: features/content/kol/src/main/java/com/tokopedia/kol/common/network/GraphqlErrorHandler.java
package com.tokopedia.kol.common.network;

import android.content.Context;

import com.tokopedia.abstraction.common.utils.network.ErrorHandler;

/**
 * @author by milhamj on 19/04/18.
 */

public class GraphqlErrorHandler {

    public static String getErrorMessage(final Context context, Throwable e) {
        if (e instanceof GraphqlErrorException) {
            return e.getLocalizedMessage();
        } else {
            return ErrorHandler.getErrorMessage(context, e);
        }
    }

}

File: features/content/kol_common/src/main/java/com/tokopedia/kolcommon/util/GraphqlErrorHandler.kt
package com.tokopedia.kolcommon.util

import android.content.Context

import com.tokopedia.abstraction.common.utils.network.ErrorHandler

/**
 * @author by milhamj on 19/04/18.
 */

object GraphqlErrorHandler {

    @JvmStatic
    fun getErrorMessage(context: Context, e: Throwable): String {
        return if (e is GraphqlErrorException) {
            e.getLocalizedMessage()
        } else {
            ErrorHandler.getErrorMessage(context, e)
        }
    }

}
